client/src/Pages/EditPage/EditPage.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import {useState, useEffect} from 'react'
+import { useNavigate } from 'react-router-dom';
+import './Edit.css'
+import Field from './Field.js'
+
+async function getUserdetails() {
+	var res = await fetch("/edit/getUser")
+	.then(res => res.json())
+	.then( data => {
+		return data
+	})
+
+}
+
+export default function EditPage(props) {
+    const [details, setDetails] = useState({});
+
+    useEffect(async () => {
+        var details = await fetch("/edit/getUser")
+        .then(res => res.json())
+        .then( data => {
+            setDetails(data)
+        })
+		//setDetails(details)
+
+    }, [])
+    
+    const userDetails = 
+        {
+        user: "User1",
+        name: "Name1",
+        money: 1234,
+        email: "email@i1232"
+    }
+    const test = [];
+    console.log(details)
+
+    for(const[key, value] of Object.entries(details)) {
+        console.log(key, value)
+        test.push([key, value]);
+    }
+
+    const userList = test.map(field => (
+        <div className={"stackIt"}>{field[0]} {field[1]} button</div>
+        
+        )
+        )
+    const userId = test.filter(field => field[0] == "user_id");
+    //console.log(userId[0][1])
+
+    const toDisplay = test.filter(field => field[0] != "user_id");
+
+    const users1 = toDisplay.map(field => (
+        <Field details={field} id={userId[0][1]}/>         
+
+    )
+    )
```

Comment:
Could you have an individual Field tag for each field you want? ex.

\<Field name="Username" id="username" value=data.username\>
\<Field name="First Name" id="first_name" value=data.first_name\>
...

I this way it'll be easier to understand from the code how the page will look and easier to know what data is being passed to each field. That way, you won't need to be converting the json to a list or anything. (Also, it's in general a good rule not to have code outside a useEffect or return statement in your Tag functions)